File: letcode/let169.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var numSize = nums.length;
    var max_v = 0, min_v = 0x7fffffff;
    for(var i = 1; i <= numSize; i ++)
    {
    	if(nums[i] < min_v)
    		min_v = nums[i];
    	if(nums[i] > max_v)
    		max_v = nums[i];

        '\n'
    }
    var a = new Array(max_v);
    for(var i = 1; i <= numSize; i ++)
    {
    	a[nums[i]]++;
    }
    for(var i = min_v; i <= max_v; i ++)
    {
    	if(a[i] >= (numSize/2+1))
    		return a[i];
    }
};
console.log(majorityElement("12333"));

File: letcode/let169.cpp
#include <iostream>
#include <cstdio>
using namespace std;
int main()
{
	int n,v,max_v = 0,min_v = 0x7fffffff;
	cin>>n;
	int a[n] = {0};
	for(int i = 1; i <= n; i ++)
	{
		cin>>v;
		if(v > max_v)
		max_v = v;
		if(v < min_v)
		min_v = v;
		a[v]++;
	}
	for(int i = min_v; i <= max_v; i ++)
	{
		//cout<<i<<":"<<a[i]<<endl;
		if(a[i] >= (n/2+1))
		{
			cout<<i<<endl;
			break;
		}
	}
	return 0;
}
